halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/providers/huaweicloud/HuaweiCloudBakeryDefaultsValidator.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2019 Huawei Technologies Co.,Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.halyard.config.validate.v1.providers.huaweicloud;
+
+import com.netflix.spinnaker.halyard.config.model.v1.node.Validator;
+import com.netflix.spinnaker.halyard.config.model.v1.providers.huaweicloud.HuaweiCloudBakeryDefaults;
+import com.netflix.spinnaker.halyard.config.model.v1.providers.huaweicloud.HuaweiCloudBaseImage;
+import com.netflix.spinnaker.halyard.config.problem.v1.ConfigProblemSetBuilder;
+import com.netflix.spinnaker.halyard.core.problem.v1.Problem;
+import com.netflix.spinnaker.halyard.core.tasks.v1.DaemonTaskHandler;
+import java.util.List;
+import lombok.EqualsAndHashCode;
+import org.springframework.util.StringUtils;
+
+@EqualsAndHashCode(callSuper = false)
+public class HuaweiCloudBakeryDefaultsValidator extends Validator<HuaweiCloudBakeryDefaults> {
+
+  @Override
+  public void validate(ConfigProblemSetBuilder p, HuaweiCloudBakeryDefaults n) {
+    DaemonTaskHandler.message(
+        "Validating "
+            + n.getNodeName()
+            + " with "
+            + HuaweiCloudBakeryDefaultsValidator.class.getSimpleName());
+
+    if ((new HuaweiCloudBakeryDefaults()).equals(n)) {
```

Comment:
I'm not sure I fully understand how this new commit fixes the bug.  Can you explain the logic behind short-circuiting here?  Thanks!

Reply:
Like google's [implementation](https://github.com/spinnaker/halyard/blob/master/halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/providers/google/GoogleBakeryDefaultsValidator.java#L48-L59), it checks whether the input is empty before doing validations. Here I did the same work, but I use `==` instead of `equals` to compare the input with an empty instance of HuaweiCloudBakeryDefaults wrongly.

Reply:
Ok, this looks good to me then, thanks for the explanation!